src/pyodide/internal/snapshot.ts

```diff
@@ -226,6 +234,7 @@ type DylinkInfo = {
   settings?: { baselineSnapshot?: boolean };
   loadOrder?: string[];
   soMemoryBases?: { [name: string]: number };
+  version?: undefined;
```

Comment:
Should this be undefined? Could it be removed?

Reply:
The point is to get type narrowing to work correctly for `type SnapshotJson = DylinkInfo | NewSnapshotJson`. In particular, if I do:
```
let a = {} as SnapshotJson;
if (a.version === 1) {
   // type of a should be narrowed to `NewSnapshotJson`
} else {
   // type of a should be narrowed to `DylinkInfo`.
}
```
If I leave this out, `a.version` is a type error and no type narrowing will happen. So it's just a type hack.